pkg/ipam/crd.go

```diff
@@ -549,14 +549,32 @@ func (n *nodeStore) isIPInReleaseHandshake(ip string) bool {
 }
 
 // allocateNext allocates the next available IP or returns an error
-func (n *nodeStore) allocateNext(allocated ipamTypes.AllocationMap, family Family) (net.IP, *ipamTypes.AllocationIP, error) {
+func (n *nodeStore) allocateNext(allocated ipamTypes.AllocationMap, family Family, owner string) (net.IP, *ipamTypes.AllocationIP, error) {
 	n.mutex.RLock()
 	defer n.mutex.RUnlock()
 
 	if n.ownNode == nil {
 		return nil, nil, fmt.Errorf("CiliumNode for own node is not available")
 	}
 
+	// check ip has already set owner
+	for ip, ipInfo := range n.ownNode.Spec.IPAM.Pool {
+		if ipInfo.Owner == owner {
+			parsedIP := net.ParseIP(ip)
+			if parsedIP == nil {
+				log.WithFields(logrus.Fields{
+					fieldName: n.ownNode.Name,
+					"ip":      ip,
+				}).Warning("Unable to parse IP in CiliumNode custom resource")
+				return nil, nil, fmt.Errorf("customer invalid ip %s for %s. ", ip, owner)
+			}
+			if DeriveFamily(parsedIP) != family {
+				return nil, nil, fmt.Errorf("customer ip for %s family is invalid %v. ", owner, family)
```

Comment:
How does this work in dual-stack setups? Shouldn't we simply ignore the wrong family here and `continue`, as we do in the regular allocation logic below?

Reply:
ok